lib/compress/zstd_compress.c

```diff
@@ -5164,11 +5164,11 @@ static size_t ZSTD_CCtx_init_compressStream2(ZSTD_CCtx* cctx,
                                              size_t inSize) {
     ZSTD_CCtx_params params = cctx->requestedParams;
     ZSTD_prefixDict const prefixDict = cctx->prefixDict;
+    if (cctx->cdict && !cctx->localDict.cdict)
+        params.compressionLevel = cctx->cdict->compressionLevel; /* let cdict take priority in terms of compression level */
```

Comment:
@senhuang42 I think this is correct, but it is really confusing to have moved this above `ZSTD_initLocalDict()`. When `ZSTD_initLocalDict()` is creates a cdict, this will never run, because necessarily `cctx->cdict == nullptr` in that case.

I believe that leaving this line in the original location (L5170), with the additional `!cctx->localDict.cdict` check will behave identically, but is much clearer.